File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1436.java
import java.util.*;

class Leetcode1436 {
    public static void main(String[] args) {
        List<List<String>> paths = new ArrayList<>();

        paths.add(List.of("London", "New York"));
        paths.add(List.of("New York", "Lima"));
        paths.add(List.of("Lima", "Sao Paulo"));
        
        System.out.println("Destination: "+ destCity(paths));
    }
    private static String destCity(List<List<String>> paths) {
        Set<String> set = new HashSet();
        for(List<String> path: paths) {
            set.add(path.get(0));
        }
        for(List<String> path: paths) {
            if(!set.contains(path.get(1))){
                return path.get(1);
            }
        }
        return null;
    }
}

File: LeetCode Questions + Tutorials/EngineeringDigest/Leetcode1436.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
function destCity(paths) {
    var set = new Set();
    for (var _i = 0, paths_1 = paths; _i < paths_1.length; _i++) {
        var path = paths_1[_i];
        set.add(path[0]);
    }
    for (var _a = 0, paths_2 = paths; _a < paths_2.length; _a++) {
        var path = paths_2[_a];
        if (!set.has(path[1])) {
            return path[1];
        }
    }
    return null;
}
;
var paths = [
    ["London", "New York"],
    ["New York", "Lima"],
    ["Lima", "Sao Paulo"]
];
console.log(destCity(paths));
